sam/pileup.go

```diff
@@ -0,0 +1,362 @@
+package sam
+
+import (
+	"fmt"
+	"github.com/vertgenlab/gonomics/chromInfo"
+	"github.com/vertgenlab/gonomics/cigar"
+	"github.com/vertgenlab/gonomics/dna"
+	"log"
+	"strings"
+)
+
+// Pile stores the number of each base observed across multiple reads.
+// The number of a particular base can be retrieved by using the dna.Base
+// as the index for Pile.Count. E.g. the number of reads with A is Count[dna.A].
+type Pile struct {
+	RefIdx   int
+	Pos      uint32
+	Count    [13]int        // Count[dna.Base] == Number of observed dna.Base
+	InsCount map[string]int // key is insertion sequence as string
+
+	touched bool // true if Count or InsCount has been modified
```

Comment:
I am not totally sure how I would use this if maintaining the code.  Maybe an additional sentence?

Reply:
added